src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -1356,12 +1362,67 @@ def reindex(self) -> None:
         ValueError
             If non-unique index and ``method`` or ``limit`` passed.
 
+        Notes
+        -----
+        If the index is lazy, ``method=nearest`` is not supported.
+
+        If the index is lazy, and duplicate values are present,
```

Comment:
We could match pandas behavior here by adding another column that indicates if there are duplicate values in the existing index, but I figured that that would add additional computation that isn't super necessary, although happy to change this if folks disagree?